DragAndDrop/DraggableView.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  Drag.swift
+//  DragAndDrop
+//
+//  Created by Stanley Chiang on 3/6/16.
+//  Copyright © 2016 SantaClaraiOSConnect. All rights reserved.
+//
+
+import UIKit
+
+protocol draggableViewDelegate {
+    func didDragToTrash() -> Bool
+}
+
+class DraggableView: UIView {
+    
+    var panGestureRecognizer:UIPanGestureRecognizer!
+    var initialPosition:CGPoint!
+    
+    var delegate:draggableViewDelegate!
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        initialPosition = frame.origin
+        panGestureRecognizer = UIPanGestureRecognizer()
+        panGestureRecognizer.addTarget(self, action: "updatePosition:")
+        addGestureRecognizer(panGestureRecognizer)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func updatePosition(sender: UIPanGestureRecognizer) {
+        let translation = sender.translationInView(self)
+
+        frame.origin.x = initialPosition.x + translation.x
+        frame.origin.y = initialPosition.y + translation.y
+        
+        let willTrash: Bool = delegate.didDragToTrash()
```

Comment:
nitpick: don't force the type to be `Bool`, let the compiler do it. (this avoids issues where the function call might return a `Bool?` but you are forcing it to `Bool` and sadness ensues.